client/my-sites/upgrades/cart/cart-coupon.jsx

```diff
@@ -77,22 +77,26 @@ module.exports = React.createClass({
 			return;
 		}
 
+		if ( this.state.hasSubmittedCoupon ) {
+			return;
+		}
+
 		return (
-			<form onSubmit={ this.applyCoupon }>
+			<div className="form">
```

Comment:
Because this PR is moving this component (based on the A/B variation, of course) to payment forms (like `client/my-sites/upgrades/checkout/credit-card-payment-box.jsx`) and [the standards prohibit a `form` inside a `form`](http://stackoverflow.com/a/555970/181497).